libraries/Microsoft.Bot.StreamingExtensions/ReceiveRequestExtensions.cs

```diff
@@ -0,0 +1,65 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.IO;
+using System.Linq;
+using System.Text;
+using Newtonsoft.Json;
+
+namespace Microsoft.Bot.StreamingExtensions
+{
+    /// <summary>
+    /// Helper methods added to the <see cref="ReceiveRequest"/> class.
+    /// </summary>
+    public static class ReceiveRequestExtensions
+    {
+        /// <summary>
+        /// Serializes the body of this <see cref="ReceiveRequest"/> as JSON.
+        /// </summary>
+        /// <typeparam name="T">The type to attempt to deserialize the contents of this <see cref="ReceiveRequest"/>'s body into.</typeparam>
+        /// <param name="request">The current instance of <see cref="ReceiveRequest"/>.</param>
+        /// <returns>On success, an object of type T populated with data serialized from the <see cref="ReceiveRequest"/> body.
+        /// Otherwise a default instance of type T.
+        /// </returns>
+        public static T ReadBodyAsJson<T>(this ReceiveRequest request)
+        {
```

Comment:
The name here is misleading. There is no "Body" every being read. Body has a very specific meaning, and it's not being used here.